### PR TITLE
approvals table: link collection names to collection view page

### LINKED ISSUE #1024: Link collection names in the Approvals table on the dashboard to the collection view page

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

Fixes #1024

I asked Amy if we should truncate collection titles at 150.  "I just checked the collections we are migrating and the longest title is 139 characters."

### Before

image

### After

image

## How was this change tested?

locally, and a unit test.

~~I wasn't sure what to test with code -- the analogous Works::LinkToShowComponent only tests that titles are truncated ... but collection titles don't need to be truncated ...~~ .

### CODE REVIEW

File: spec/features/mediated_deposit_versioning_spec.rb [hunk 1]
```diff
@@ -37,6 +37,13 @@
       expect(page).to have_content 'You have successfully submitted your deposit'
       click_link 'Return to dashboard'
 
+      # check collection link in approvals table
+      within_table('Approvals') do
```
Comment (jcoyne) [inline spec/features/mediated_deposit_versioning_spec.rb:41]: I'd prefer that we keep this test focused on doing a mediated deposit.  Can this be in a component test instead?

Comment (ndushay) [reply spec/features/mediated_deposit_versioning_spec.rb:41]: FYI, as I mentioned, the analogous Works::LinkToShowComponent only tests that titles are truncated, nothing about the link.

**Comment by ndushay** [comment]
@jcoyne ready for re-review.

File: app/components/collections/link_to_show_component.html.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+<td class="collection-title"><%= link %></td>
```
Comment (jcoyne) [inline app/components/collections/link_to_show_component.html.erb:1]: I'm not terribly fond of having the `<td>` in this component, as that limits where we're able to use it.  Can that be moved back into ApprovalsComponent?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/components/dashboard/approvals_component.html.erb [hunk 3]
```diff
@@ -16,7 +16,7 @@
       <tr>
         <%= render Works::LinkToShowComponent.new(work_version: submission) %>
```
Comment (ndushay) [inline app/components/dashboard/approvals_component.html.erb:17]: I heard what you are saying about the <td> but look at line 17 here ...

**Comment by ndushay** [comment]
back to you, @jcoyne

**Review by jcoyne — APPROVED** [review]

**Pull request merged**